### PR TITLE
[5.0] Fix "getCountForPagination" For Grouped Query

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Moved over from #8113. I've just committed it all as @codelight89.

// cc @fcjailybo, @JoostK, @codelight89, @jarektkaczyk

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1420,16 +1420,28 @@ public function simplePaginate($perPage = 15, $columns = ['*'])
...
 		$this->restoreFieldsForCount();
 
+		if (isset($results[0]))
+		{
+			if (isset($this->groups)) return count($results);
+
+			return (int) array_change_key_case((array) $results[0])['aggregate'];
+		}
+
 		return $total;
```
Comment (JoostK) [inline src/Illuminate/Database/Query/Builder.php]: This line should be removed. Moreover, the method may now return null. I would change it all as follows:

```php
// ...
$this->restoreFieldsForCount();

if (isset($this->groups)) return count($results);

return isset($results[0]) ? (int) array_change_key_case((array) $results[0])['aggregate'] : 0;
```

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: I'll make this change, and commit as your user.

Comment (JoostK) [inline src/Illuminate/Database/Query/Builder.php]: Wait, what? How'd you do that?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: `git commit --author`

**Comment by simonfranz** [comment]
+1

**Review by ravi-azova — COMMENTED** [review]
Great !

**Pull request merged**